src/execution_plan/execution_plan_build/build_match_op_tree.c

```diff
@@ -109,10 +109,10 @@ static void _ExecutionPlan_ProcessQueryGraph
 			}
 
 			if(edge && (QGEdge_VariableLength(edge) || !QGEdge_SingleHop(edge))) {
-				// edge is part of a shortest-path
-				// MATCH allShortestPaths((a)-[*..]->(b))
-				// validate both edge ends are bounded
 				if(QGEdge_IsShortestPath(edge)) {
+					// edge is part of a shortest-path
```

Comment:
don't we miss check QGEdge_GhostEdge in line 111

Reply:
no,
if `edge` is not NULL AND
`QGEdge_VariableLength(edge)` is `FALSE` AND
`!QGEdge_SingleHop(edge)` returns `TRUE`

This means we're dealing with a 0 length edge.

edge is referenced, not variable length and not single hop.
this means constant length which is not of length 1, could be 0, 2, 3....
in which case we need to use the `CondVarLenTraverseOp` operation